presto-spi/src/main/java/io/prestosql/spi/connector/ProjectionApplicationResult.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.connector;
+
+import io.prestosql.spi.expression.ConnectorExpression;
+import io.prestosql.spi.type.Type;
+
+import java.util.List;
+
+public class ProjectionApplicationResult<T>
+{
+    private final T handle;
+    private final List<ConnectorExpression> newProjections;
+    private final List<Assignment> assignments;
+
+    public ProjectionApplicationResult(T handle, List<ConnectorExpression> newProjections, List<Assignment> assignments)
+    {
+        this.handle = handle;
+        this.newProjections = newProjections;
+        this.assignments = assignments;
+    }
+
+    public T getHandle()
+    {
+        return handle;
+    }
+
+    public List<ConnectorExpression> getNewProjections()
+    {
+        return newProjections;
+    }
+
+    public List<Assignment> getAssignments()
+    {
+        return assignments;
+    }
+
+    public static class Assignment
+    {
+        private final String variable;
+        private final ColumnHandle column;
```

Comment:
why not use `ColumnHandles` directly in connector expressions?
Do you think there will be other variables than `ColumnHandles`?

Reply:
Yes, later we’ll have lambda expressions and possibly a way to do “assignments” (as in let-assignments)